apps/src/blockly/customBlocks/googleBlockly/mutators/procedureDefMutator.js

```diff
@@ -159,6 +160,7 @@ export const procedureDefMutator = {
     setBlockDescription(this, state['description']);
     this.doProcedureUpdate();
     this.setDeletable(state['initialDeleteConfig'] === false ? false : true);
+    this.setMovable(state['initialMoveConfig'] === false ? false : true);
```

Comment:
Hopefully no negative side-effects to supporting `initialMoveConfig` the same way we already support `initialDeleteConfig`.

Reply:
I'd defer to @mikeharv here

Reply:
It looks like you're reading the movable state early on and then just reapply it here, if there was a valid value, defaulting to movable. i don't have the full context for why this is needed, but it seems reasonable on its own to me!

Reply:
Oh wait, I do have context :D

I just didn't realize you all were using procedures in Music Lab already. The modal editor forces procedure blocks to be immovable, but if that wasn't explicitly the intention for the blocks if they were on the main workspace, we'd want to preserve that original intention. Sound right?

Reply:
In this case, we want to provide a function block in the starter code, and we want it to be immovable and undeletable, and it seemed [this](https://github.com/code-dot-org/code-dot-org/pull/55648/files#diff-7be90885a88d04290de0cfa9cd2ba9eb24bc1860baadc2d47973ce091e0971bcR31-R32) was the only way to do it, but required adding this new code.